File: java-compat/Manual/simpleFilesystem/Filesystem/FileI.java
// **********************************************************************
//
// Copyright (c) 2003-2016 ZeroC, Inc. All rights reserved.
//
// **********************************************************************

package Filesystem;

public class FileI extends _FileDisp
{
    // FileI constructor

    public
    FileI(Ice.Communicator communicator, String name, DirectoryI parent)
    {
        _name = name;
        _parent = parent;

        assert(_parent != null);

        //
        // Create an identity
        //
        _id = new Ice.Identity();
        _id.name = java.util.UUID.randomUUID().toString();
    }

    // Slice Node::name() operation

    public String
    name(Ice.Current current)
    {
        return _name;
    }

    // Slice File::read() operation

    public String[]
    read(Ice.Current current)
    {
        return _lines;
    }

    // Slice File::write() operation

    public void
    write(String[] text, Ice.Current current)
        throws GenericError
    {
        _lines = text;
    }

    public void
    activate(Ice.ObjectAdapter a)
    {
        NodePrx thisNode = NodePrxHelper.uncheckedCast(a.add(this, _id));
        _parent.addChild(thisNode);
    }

    private String _name;
    private DirectoryI _parent;
    private Ice.Identity _id;
    private String[] _lines;
}

File: csharp/Manual/simpleFilesystem/Server/FileI.cs
// Copyright (c) ZeroC, Inc.

using Filesystem;
using System.Diagnostics;

internal class FileI : FileDisp_
{
    // FileI constructor
    public FileI(string name, DirectoryI parent)
    {
        _name = name;
        _parent = parent;

        Debug.Assert(_parent != null);

        // Create an identity
        _id = new Ice.Identity();
        _id.name = System.Guid.NewGuid().ToString();
    }

    // Slice Node::name() operation
    public override string name(Ice.Current current)
    {
        return _name;
    }

    // Slice File::read() operation
    public override string[] read(Ice.Current current)
    {
        return _lines;
    }

    // Slice File::write() operation
    public override void write(string[] text, Ice.Current current = null)
    {
        _lines = text;
    }

    // Add servant to ASM and parent's _contents map.
    public void Activate(Ice.ObjectAdapter a)
    {
        var thisNode = NodePrxHelper.uncheckedCast(a.add(this, _id));
        _parent.AddChild(thisNode);
    }

    private string _name;
    private readonly DirectoryI _parent;
    private Ice.Identity _id;
    private string[] _lines;
}
